tests/test_other.py

```diff
@@ -9639,3 +9639,26 @@ def test_run_wasi_sdk_output(self):
     self.compile_with_wasi_sdk(path_from_root('tests', 'hello_world.c'), 'hello.wasm')
     self.run_process([EMCC, '--post-link', '-sPURE_WASI', 'hello.wasm'])
     self.assertContained('hello, world!', self.run_js('a.out.js'))
+
+  def run_wasi_test_suite_test(self, name):
+    wasm = path_from_root('tests', 'third_party', 'wasi-test-suite', name + '.wasm')
+    if not os.path.exists(wasm):
+      self.fail('wasi-test-suite not found; run `git submodule update --init`')
+    self.run_process([EMCC, '--post-link', '-g', '-sPURE_WASI', '-lnodefs.js', '-lnoderawfs.js', wasm, '-o', name + '.js'])
```

Comment:
It's a little funny to set `PURE_WASI` and also NODERAWFS here... I guess this is the "normal" mode wasi works in, while emscripten's default is to sandbox the filesystem entirely. Anyhow, just a thought, I don't have an idea for an improvement.

Reply:
I think it makes sense.   PUSE_WASI is a description of the way the wasm file interacts with JS, it describes the shape of the syscall layer.    noderawfs is effecively a plugin the JS-based filesystem code right?  It should work just fine whatever the shape of the syscall layer that goes from wasm to JS.